x-pack/legacy/plugins/maps/public/layers/layer.js

```diff
@@ -251,6 +251,24 @@ export class AbstractLayer {
     return this._source.renderSourceSettingsEditor({ onChange });
   };
 
+  getPrevRequestToken(dataId) {
+    const prevDataRequest = this.getDataRequest(dataId);
+    if (!prevDataRequest) {
+      return;
+    }
+
+    return prevDataRequest.getRequestToken();
+  }
+
+  getInFlightRequestTokens() {
+    if (!this._dataRequests || this._dataRequests.length === 0) {
```

Comment:
You don't need the second condition here, but I'm fine with leaving it if you'd prefer